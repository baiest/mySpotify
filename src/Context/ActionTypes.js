export const SET_USER = 'SET_USER'
export const SET_CURRENT_TRACK = 'SET_CURRENT_TRACK'
export const CLOSE_PLAYER = 'CLOSE_PLAYER'