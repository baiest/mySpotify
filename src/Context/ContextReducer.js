import {
  CLOSE_PLAYER,
  SET_CURRENT_TRACK,
  SET_USER
} from './ActionTypes'

export const initialState = {
  user: null,
  currentTrack: null
}


const reducerOptions = (state, payload) => ({
  [SET_USER]: {...state, user: payload },
  [SET_CURRENT_TRACK]: {...state, currentTrack: payload },
  [CLOSE_PLAYER]: {...state, currentTrack: null }
})

export const contextReducer = (state, action) => {
  try {
    const options = reducerOptions(state, action.payload)
    return options[action.type]
  } catch (error) {
    console.error('Accion no encontrada: ', action.type, 'Error: ', error)
    return state
  }
}

export const setUser = (user) => ({
  type: SET_USER,
  payload: user
})

export const setTrack = (idTrack) => ({
  type: SET_CURRENT_TRACK,
  payload: idTrack
})

export const closePlayer = () => ({ type: CLOSE_PLAYER })