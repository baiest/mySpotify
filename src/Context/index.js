import React from "react"

export const AppContext = React.createContext()

export const ContextProvider = ({ children, reducer, initialState }) => {
  const [state, dispatch] = React.useReducer(reducer, initialState)
  return (
    <AppContext.Provider value={[state, dispatch]}>
      {children}
    </AppContext.Provider>
  )
}
