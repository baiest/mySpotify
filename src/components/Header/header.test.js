/*eslint-disable*/
import React from "react"
import renderer from "react-test-renderer"
import { ContextProvider } from "../../Context"
import { contextReducer, initialState } from "../../Context/ContextReducer"
import { Header } from "./index"
describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ContextProvider initialState={initialState} reducer={contextReducer}>
          <Header siteTitle="MySpotify" />
        </ContextProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
