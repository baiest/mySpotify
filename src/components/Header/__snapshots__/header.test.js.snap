// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <h1>
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      MySpotify
    </a>
  </h1>
</header>
`;
